mock-data-provider-plugin/src/main/java/io/mosip/certify/mockdataprovider/integration/service/MockDataProviderPlugin.java

```diff
@@ -0,0 +1,160 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.CacheManager;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import javax.crypto.Cipher;
+import java.security.Key;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+@Component
+@Slf4j
+public class MockDataProviderPlugin implements DataProviderPlugin {
+    private static final String AES_CIPHER_FAILED = "aes_cipher_failed";
+    private static final String NO_UNIQUE_ALIAS = "no_unique_alias";
+    private static final String USERINFO_CACHE = "userinfo";
+
+    @Autowired
+    private CacheManager cacheManager;
+
+    @Autowired
+    private KeyStore keyStore;
+
+    @Autowired
+    private KeymanagerDBHelper dbHelper;
+
+    @Value("${mosip.certify.mock.authenticator.get-identity-url}")
+    private String getIdentityUrl;
+
+    @Value("${mosip.certify.cache.security.secretkey.reference-id}")
+    private String cacheSecretKeyRefId;
+
+    @Value("${mosip.certify.cache.security.algorithm-name}")
+    private String aesECBTransformation;
+
+    @Value("${mosip.certify.cache.secure.individual-id}")
+    private boolean secureIndividualId;
+
+    @Value("${mosip.certify.cache.store.individual-id}")
+    private boolean storeIndividualId;
+
+    @Value("#{${mosip.certify.mock.vciplugin.vc-credential-contexts:{'https://www.w3.org/2018/credentials/v1','https://schema.org/'}}}")
+    private List<String> vcCredentialContexts;
+
+    private static final String ACCESS_TOKEN_HASH = "accessTokenHash";
+
+    public static final String UTC_DATETIME_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+
+    public static final String CERTIFY_SERVICE_APP_ID = "CERTIFY_SERVICE";
+
+    @Override
+    public Map<String, Object> fetchData(Map<String, Object> identityDetails) throws DataProviderExchangeException {
+        OIDCTransaction transaction = getUserInfoTransaction(identityDetails.get(ACCESS_TOKEN_HASH).toString());
+        Map<String, Object> formattedMap = null;
+        try {
+            formattedMap = getIndividualData(transaction);
+        } catch (Exception e) {
+            log.error("Unable to get KYC exchange data from MOCK", e);
+        }
+
+        Map<String, Object> verCredJsonObject = new HashMap<>();
+        verCredJsonObject.put("@context", vcCredentialContexts);
+        verCredJsonObject.put("type", Arrays.asList("VerifiableCredential", "MockVerifiableCredential"));
+        verCredJsonObject.put("id", "urn:uuid:3978344f-8596-4c3a-a978-8fcaba3903c5");
+        verCredJsonObject.put("issuer", "did:example:123456789");
+        verCredJsonObject.put("validFrom", getUTCDateTime());
+        verCredJsonObject.put("credentialSubject", formattedMap);
+
+        return verCredJsonObject;
```

Comment:
this isn't required, the DataProviderPlugin interface returns the raw data which the template can understand